source/main.cpp

```diff
@@ -67,18 +86,22 @@ int main(int argc, char** argv)
 
     sh3_gl::program::load_error err;
     sh3_gl::program test("test", err);
-    sh3_gl::vao<MyAttributes> myVao;
 
-    myVao.Bind();
-    sh3_gl::buffer_object& verts = myVao[MyAttributes::VERTEX_SLOT];
+    using Triangle = sh3_gl::vao<TriangleAttributes>;
+
+    Triangle triVao;
+
+    triVao.Bind();
+
+    sh3_gl::buffer_object& verts = triVao[Triangle::Slot::VERTEX];
     verts.BufferData(g_vertex_buffer_data, sizeof(g_vertex_buffer_data), GL_STATIC_DRAW);
-    myVao.SetDataLocation(MyAttributes::VERTEX_SLOT, sh3_gl::vao<MyAttributes>::DataType::FLOAT, 3, 0, 0);
+    triVao.SetDataLocation(Triangle::Slot::VERTEX, Triangle::DataType::FLOAT, 3, 0, 0);
```

Comment:
So `Triangle::Slot::VERTEX` needs to be mapped to the same position as the one in `sh3_gl::vao_target_array<Slot> Targets` yes?